src/screencastPanel.ts

```diff
@@ -0,0 +1,100 @@
+import { encodeMessageForChannel, WebSocketEvent } from './common/webviewEvents';
+import { PanelSocket } from './panelSocket';
+import { ScreencastView } from './screencast/view';
+import {
+    SETTINGS_STORE_NAME,
+    SETTINGS_SCREENCAST_WEBVIEW_NAME,
+} from './utils';
+
+export class ScreencastPanel {
+    private static instance: ScreencastPanel | undefined;
+    private readonly context: vscode.ExtensionContext;
+    private readonly extensionPath: string;
+    private readonly panel: vscode.WebviewPanel;
+    private targetUrl: string
+    private panelSocket: PanelSocket;
+
+    private constructor(
+        panel: vscode.WebviewPanel,
+        context: vscode.ExtensionContext,
+        targetUrl: string) {
+        this.panel = panel;
+        this.context = context;
+        this.targetUrl = targetUrl;
+        this.extensionPath = this.context.extensionPath;
+
+        this.panelSocket = new PanelSocket(this.targetUrl, (e, msg) => this.postToWebview(e, msg));
+        this.panelSocket.on('websocket', () => this.onSocketMessage());
+        this.panelSocket.on('close', () => this.onSocketClose());
+
+        // Handle closing
+        this.panel.onDidDispose(() => {
+            this.dispose();
+            this.panelSocket.dispose();
+        }, this);
+
+        // Handle view change
+        this.panel.onDidChangeViewState(_e => {
+            if (this.panel.visible) {
+                this.update();
+            }
+        }, this);
+
+        // Handle messages from the webview
+        this.panel.webview.onDidReceiveMessage(message => {
+            this.panelSocket.onMessageFromWebview(message);
+        }, this);
+    }
+
+    dispose(): void {
+        ScreencastPanel.instance = undefined;
+
+        this.panel.dispose();
+        this.panelSocket.dispose();
+    }
+
+    private onSocketMessage() {
+        // TODO: Handle message
+    }
+
+    private onSocketClose() {
+        this.dispose();
+    }
+
+    private update() {
+        // Check to see which version of devtools we need to launch
+        this.panel.webview.html = this.getHtmlForWebview();
+    }
+
+    private postToWebview(e: WebSocketEvent, message?: string) {
+        encodeMessageForChannel(msg => this.panel.webview.postMessage(msg) as unknown as void, 'websocket', { event: e, message });
+    }
+
+    private getHtmlForWebview() {
+        const inspectorPath = vscode.Uri.file(path.join(this.extensionPath, 'out/screencast', 'screencast.bundle.js'));
+        const inspectorUri = this.panel.webview.asWebviewUri(inspectorPath);
+		const codiconsUri = this.panel.webview.asWebviewUri(vscode.Uri.joinPath(this.context.extensionUri, 'node_modules', '@vscode/codicons', 'dist', 'codicon.css'));
+        const cssPath = this.panel.webview.asWebviewUri(vscode.Uri.joinPath(this.context.extensionUri, 'out/screencast', 'view.css'));
+        const view = new ScreencastView(this.panel.webview.cspSource, cssPath, codiconsUri, inspectorUri);
```

Comment:
nit: since it looks like we create a new view, then call render only once, wdyt about trying to pull the member variables out of ScreencastView and making all of the methods static?

Reply:
I guess we could transform the constructor in a static render method, but I don't see a lot of advantages on it. Keeping it encapsulated in the constructor feels right in this case as the object should not exist without the properties.